### PR TITLE
Mpuncel/pod cluster labels

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/kp/pcstore/consul_store.go [hunk 1]
```diff
@@ -75,7 +81,18 @@ func (s *consulStore) Create(
 	if !success {
 		return fields.PodCluster{}, util.Errorf("Could not set pod cluster at path '%s'", key)
 	}
-	return pc, nil
+
+	// Should we delete the PC if the labels fail?
```
Comment (anthonybishopric) [inline pkg/kp/pcstore/consul_store.go]: Yes, lets

File: pkg/kp/rcstore/consul_store.go [hunk 2]
```diff
@@ -85,6 +85,8 @@ func NewConsul(client *api.Client, retries int) *consulStore {
 
 func (s *consulStore) Create(manifest pods.Manifest, nodeSelector klabels.Selector, podLabels klabels.Set) (fields.RC, error) {
 	rc, err := s.innerCreate(manifest, nodeSelector, podLabels)
+
+	// TODO: measure whether retries are is important in practice
```
Comment (anthonybishopric) [inline pkg/kp/rcstore/consul_store.go]: Good candidate for a metric

File: pkg/kp/pcstore/consul_store.go [hunk 3]
```diff
@@ -75,9 +82,25 @@ func (s *consulStore) Create(
 	if !success {
 		return fields.PodCluster{}, util.Errorf("Could not set pod cluster at path '%s'", key)
 	}
+
+	// Should we delete the PC if the labels fail?
+	err = s.setLabelsForPC(pc)
+	if err != nil {
+		return fields.PodCluster{}, s.Delete(pc.ID)
```
Comment (anthonybishopric) [inline pkg/kp/pcstore/consul_store.go]: You should still return `err` from the label set failure.

Comment (mpuncel) [inline pkg/kp/pcstore/consul_store.go]: yeah, although now i don't know what to do if delete fails. I guess i'll add a logger and log it. we should probably throw exception reporting integration on these stores as well

**Comment by anthonybishopric** [comment]
:+1:

File: pkg/kp/pcstore/consul_store.go [hunk 4]
```diff
@@ -75,9 +82,27 @@ func (s *consulStore) Create(
 	if !success {
 		return fields.PodCluster{}, util.Errorf("Could not set pod cluster at path '%s'", key)
 	}
+
+	// Should we delete the PC if the labels fail?
```
Comment (petertseng) [inline pkg/kp/pcstore/consul_store.go]: probably rm comment

**Pull request merged**